### PR TITLE
change return value to BoxFile for UploadNewVersionUsingSessionAsync

### PR DESCRIPTION
**Pull request by carycheng**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @carycheng, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: Box.V2.Test/BoxFilesManagerTest.cs [hunk 1]
```diff
@@ -8,6 +8,10 @@
 using System.Net.Http.Headers;
 using System.IO;
 using Newtonsoft.Json;
+using Box.V2.Utility;
+using System.Reflection;
+using System.Diagnostics;
+using System.Text;
```
Comment (mattwiller) [inline Box.V2.Test/BoxFilesManagerTest.cs]: You may want to remove the ones of these that you're no longer using

File: Box.V2.Test/BoxFilesManagerTest.cs [hunk 2]
```diff
@@ -21,6 +25,36 @@ public BoxFilesManagerTest()
...
+                {
+                    Status = ResponseStatus.Success,
+                    ContentString = responseString
+                }));
+
+            var fakeStream = new Mock<System.IO.Stream>();
+        
+            
+           BoxFile f = await _filesManager.UploadNewVersionUsingSessionAsync(fakeStream.Object, "fakeId", null, null, null);
+           Assert.AreEqual("file", f.Type);
```
Comment (mattwiller) [inline Box.V2.Test/BoxFilesManagerTest.cs]: Maybe assert about the ID as well?

**Comment by mattwiller** [comment]
Looks good to me!  Just a few small changes to clean up

**Pull request merged**